File: stack/stack_using_LinkedList.c
/* A basic Stack implementation in C using linkedlist, with the following functionalities implemented:
	-Push operation 
	-Pop operation
	-Display stack
*/

#include<stdio.h>
#include<stdlib.h>

// Node, the basic data structure in single linkedlist
typedef struct node{
	int data;
	struct node *next;
}node;

//Push operation in the stack , which inserts the elements in the stack
void push(node **top, int data){
	node *newNode = (node *)malloc(sizeof(node));
	if(!newNode){                        // Condition for, Stack is full
		printf("\nSTACK OVERFLOW! HEAP FULL\n");
		return;
	}
	newNode->data = data;
	newNode->next = *top;
	*top = newNode;
}

//Pop operation in the stack , whitch delete the elements from the stack
int pop(node **top){
	if(!*top){                          // Condition for, Stack is empty
		printf("\nSTACK UNDERFLOW\n");
		return 0;
	}
	node *temp = *top;
	*top = (*top)->next;
	int data = temp->data;
	free(temp);                         // Free up the memory
	return data;
}

//Display Stack
void displayStack(node *head){
	node *temp = head;
	if(temp){
		printf("%d ",temp->data);
		displayStack(temp->next);
	}
}
int main(){
	node *stack=NULL;
	push(&stack,1);
	push(&stack,2);
	push(&stack,3);
	push(&stack,5);
	displayStack(stack);
	printf("\n");
	int data = pop(&stack);
	displayStack(stack);
	printf("\n");
	return 0;
}

File: stack/stack_using_LinkedList.py
''' A basic Stack implementation in Python using Linked List, 
    with the following functionalities implemented:
	-Push operation 
	-Pop operation
	-Display stack
'''

class Node:
    def __init__(self,data): 
        self.data = data 
        self.next = None
      
class Stack:
    def __init__(self): 
        self.head = None
    
    def isempty(self): 
        
        if self.head == None: 
            return True
        else: 
            return False
    
    def insertElement(self,data): 
          
        if self.head == None: 
            self.head=Node(data) 
              
        else: 
            newnode = Node(data) 
            newnode.next = self.head 
            self.head = newnode 
        
        print("\n")
      
    def deleteElement(self): 
          
        if self.isempty(): 
            return None
              
        else:
            poppednode = self.head 
            self.head = self.head.next
            poppednode.next = None
            return poppednode.data 
        
    def printStack(self): 
          
        iternode = self.head 
        if self.isempty(): 
            print("Stack Underflow") 
          
        else: 
            while(iternode != None): 
                print(iternode.data,end = " ")
                iternode = iternode.next
                if iternode is not None:
                    print("->", end=" ")
            return
        print("\n")
        
        
if __name__ == "__main__":
    lst = Stack()
    
    while True:
        print("Select the any option")
        print("[1] Insert Element")
        print("[2] Delete Element")
        print("[3] Print List")
        print("[4] Exit")
        print("\n")
        i = int(input())
        print("\n")

        if i == 1:
            lst.insertElement(input("Enter data you want to insert: "))
        elif i == 2:
            lst.deleteElement()
        elif i == 3:
            lst.printStack()
        elif i == 4:
            break
        